scripts/test-end-to-end.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+# Create miden-client.toml file
+# Needed for the Miden client to work properly
+{
+    echo "[rpc]"
+    echo "endpoint = { protocol = \"http\", host = \"localhost\", port = 57291 }"
+    echo ""
+    echo "[store]"
+    echo "database_filepath = \"store.sqlite3\""
+} > miden-client.toml
+
+# Test flow 1
+
+# Create accounts
+miden-client account new basic-immutable
+miden-client account new fungible-faucet --token-symbol JACK --decimals 3 --max-supply 1000
+
+# Capture IDs
+basic_account=$(miden-client account -l | grep -oE ' 0x[a-fA-F0-9]{16} ' | head -n 1)
+faucet_account=$(miden-client account -l | grep -oE ' 0x[a-fA-F0-9]{16} ' | head -n 2 | tail -n 1)
+
+# Sync client and create tx between faucet and account
+miden-client sync
+miden-client tx new mint $basic_account $faucet_account 777
+sleep 15
```

Comment:
why is this needed, can we get rid of this? `sleep` in tests is to me a big no-no.

Reply:
The sleep is needed to leave some time for the rollup to include the transaction in a block and the block to be applied to the db. If not then the note does not have a commit height and it cannot be consumed by the Miden client. 

Do you see a better solution than sleeping?

Reply:
Polling would be the best option here. Probably the miden-client should have a command similar to `miden-client wait-for-note <note_id>` where the client would poll until the note_id is available.

Reply:
A couple of other ways to implement this now:

- Wait for two blocks to be created before moving on (`miden-client info` returns the current block number; should be fairly easy to parse the response and sync until you are one or two blocks above the original count)
- Wait until the transaction has been committed (although there is not any super simple way to programmatically get this from the client, but you can use `tx list` and check that the transaction ID is now committed)